services/src/main/java/org/keycloak/broker/oidc/AbstractOAuth2IdentityProvider.java

```diff
@@ -427,7 +427,11 @@ protected JsonWebToken generateToken() {
         jwt.type(OAuth2Constants.JWT);
         jwt.issuer(getConfig().getClientId());
         jwt.subject(getConfig().getClientId());
-        jwt.audience(getConfig().getTokenUrl());
+        String audience = getConfig().getClientAssertionAudience();
+        if (audience == null || audience.isEmpty()) {
```

Comment:
Better use `org.keycloak.utils.StringUtil#isNotBlank` to make sure the string is also trimmed before comparison.